ui/static/ui/js/learning_resources.jsx

```diff
@@ -120,8 +127,65 @@ define('learning_resources', [
         _.filter(e.target.options, function(option) {
           return option.selected && option.value !== null;
         }), 'value');
-
-      this.props.setValues(this.props.selectedVocabulary.slug, selectedValues);
+      // check if the current vocabulary allows free tagging and in case add
+      // the new tags before proceeding
+      if (this.props.selectedVocabulary.vocabulary_type === 'f') {
+        // extract the list of terms not in the vocabulary existing terms
+        var termsSlugs = _.pluck(this.props.selectedVocabulary.terms, 'slug');
+        var termsToCreate = _.filter(selectedValues, function (slug) {
+          return _.indexOf(termsSlugs, slug) === -1;
+        });
+        // if there are no new terms to create, set the state
+        if (!termsToCreate.length) {
+          this.props.setValues(
+            this.props.selectedVocabulary.slug, selectedValues
+          );
+        } else {
```

Comment:
I was thinking about it and you're right in that this case wouldn't happen since we're only inside this branch if there's at least one term to create.

I'm still uneasy about the `splice` below and how the new terms are mixed with the old slug values in the same array. I'd prefer if we remove the new terms from `selectedValues` after `termsToCreate` is set, and do the single `setValues` call after all POSTS complete where we add the new terms with the proper slugs. But I don't see a logical error in your approach, just my opinion on code cleanliness so feel free to go either way